Firestore/core/src/model/value_util.cc

```diff
@@ -522,7 +654,32 @@ Message<google_firestore_v1_Value> MaxValue() {
 }
 
 bool IsMaxValue(const google_firestore_v1_Value& value) {
-  return value == *MaxValue();
+  bool is_max =
+      (value.which_value_type == google_firestore_v1_Value_map_value_tag);
+  if (!is_max) {
```

Comment:
Probably just a personal preference, but I think this would be easier to read if you directly used the conditions in the if statement rather than `is_max`.